app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comment, class_name: 'Comment', foreign_key: 'author_id', dependent: :destroy
+  has_many :like, class_name: 'Like', foreign_key: 'author_id', dependent: :destroy
+
+  after_save :update_post_counter
+
+  def update_post_counter
```

Comment:
- [ ] To ensure that the `update_post_counter` is properly accessed kindly mark it as _private_